cdap-client/src/main/java/co/cask/cdap/client/StreamClient.java

```diff
@@ -253,10 +254,10 @@ public void setTTL(String streamId, long ttlInSeconds)
    * @return list of {@link StreamRecord}s
    * @throws IOException if a network error occurred
    */
-  public List<StreamRecord> list() throws IOException, UnauthorizedException {
+  public List<StreamSpecification> list() throws IOException, UnauthorizedException {
     URL url = config.resolveNamespacedURLV3("streams");
     HttpResponse response = restClient.execute(HttpMethod.GET, url, config.getAccessToken());
-    return ObjectResponse.fromJsonBody(response, new TypeToken<List<StreamRecord>>() { }).getResponseObject();
+    return ObjectResponse.fromJsonBody(response, new TypeToken<List<StreamSpecification>>() { }).getResponseObject();
```

Comment:
Oh, looks like there's a typo in the handler code then, b/c I see it's return `specs` rather than `result` which actually isn't returned. https://github.com/caskdata/cdap/blob/develop/cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricDataHttpHandler.java#L76